mesonbuild/modules/pkgconfig.py

```diff
@@ -319,9 +319,13 @@ def _make_relative(self, prefix, subdir):
         prefix = PurePath(prefix)
         subdir = PurePath(subdir)
         try:
-            return subdir.relative_to(prefix).as_posix()
+            libdir = subdir.relative_to(prefix)
         except ValueError:
-            return subdir.as_posix()
+            libdir = subdir
+        if libdir.is_absolute():
+            return libdir.as_posix()
+        else:
+            return f'${{prefix}}/{libdir.as_posix()}'
```

Comment:
This should be equivalent to `Path(prefix).joinpath(libdir).as_posix()`

Reply:
No, because this is a pkg-config file so `${prefix}` is a string literal and not a python variable.

Reply:
Oh, right. So `return Path('${prefix}').joinpath(libdir).as_posix()` ?

Reply:
What's the advantage of that over the current method?

Reply:
I find it dangerous and useless to use a f-string in this case, `'${prefix}/' + libdir.as_posix()` does the same. Less confusing than the double brackets.

Reply:
@keszybz IMHO it's not worth going through a Path() object just to literaly put a '/' between 2 strings. In this case we explicitly don't want a platform specific path.

Reply:
Fair enough about the f-string confusion.

Reply:
Please note the comment is attached to 4 lines. `.joinpath` does what the conditional does.

Reply:
ah, I see, makes more sense indeed. I'm fine either way, for small nitpicking like that I leave it to the author to pick their preferred way :)

Reply:
I have been trying to decide whether using pathlib on a string literal `${prefix}` feels better or worse than having more lines of code...